### PR TITLE
Refactor PropertySyntaxVisitor

### PR DESCRIPTION
**Pull request by fdiaz**
**This PR is easier to review commmit-by-commit**

I'm going to be building on top of the `PropertySyntaxVisitor` to get properties information onto types, so I wanted to refactor the `TypeSyntaxVisitor` and split the `PropertySyntaxVisitor` into its own file that we can build on top of.

### CODE REVIEW

File: Sources/SwiftInspectorVisitors/PropertySyntaxVisitor.swift [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+import Foundation
+import SwiftSyntax
+
+// MARK: - PropertySyntaxVisitor
+
+public final class PropertySyntaxVisitor: SyntaxVisitor {
```
Comment (fdiaz) [inline Sources/SwiftInspectorVisitors/PropertySyntaxVisitor.swift:28]: This was copy pasted from https://github.com/fdiaz/SwiftInspector/pull/96/commits/fef31908a664e3c6e002ab5eea64975e1745f89a#diff-d997817cea70a08592f0e08e023e249c9c161b083630f0fdf001169789a56d95L106

File: Sources/SwiftInspectorAnalyzers/PropertyAnalyzer.swift [hunk 2]
```diff
@@ -29,7 +29,7 @@ extension StandardAnalyzer {
   public func analyzeProperties(
     in fileURL: URL,
     for typeName: String) throws
-  -> Set<PropertyData>?
+  -> Set<PropertyInfo>?
```
Comment (fdiaz) [inline Sources/SwiftInspectorAnalyzers/PropertyAnalyzer.swift:32]: Following the pattern in other files.

File: Sources/SwiftInspectorVisitors/Tests/PropertySyntaxVisitorSpec.swift [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+// SOFTWARE.
+
+import Foundation
+import SwiftInspectorTestHelpers
+import Quick
+import Nimble
+
+@testable import SwiftInspectorVisitors
+
+final class PropertySyntaxVisitorSpec: QuickSpec {
```
Comment (fdiaz) [inline Sources/SwiftInspectorVisitors/Tests/PropertySyntaxVisitorSpec.swift:30]: This is a copy/paste from https://github.com/fdiaz/SwiftInspector/pull/96/commits/0050b8d6add3d106a627d104742f21ad9ccf4379#diff-982ee3afe15209354e026059da96ee6332d9c4973e7b431ff9c37bc741d25e25L334

**Review by dfed — APPROVED** [review]
thanks!

**Pull request merged**